src/applications/edu-benefits/1995/pages/stem.js

```diff
@@ -9,6 +12,82 @@ const {
 } = fullSchema1995.properties;
 
 export const uiSchema = {
+  'ui:title': 'Rogers STEM Scholarship',
+  'view:rogersStemScholarshipInfo': {
+    'ui:description': rogersStemScholarshipInfo,
+  },
+  isEdithNourseRogersScholarship: {
+    'ui:title': 'Are you applying for the Rogers STEM Scholarship?',
+    'ui:widget': 'yesNo',
+  },
+  isEnrolledStem: {
+    'ui:title':
+      'Are you enrolled in a science, technology, engineering, or math (STEM) undergraduate degree program?',
+    'ui:widget': 'yesNo',
+    'ui:required': form => form.isEdithNourseRogersScholarship,
+    'ui:options': {
+      expandUnder: 'isEdithNourseRogersScholarship',
+    },
+  },
+  'view:exhaustionOfBenefits': {
+    'ui:title':
+      'Have you used all of your Post-9/11 GI Bill or Fry Scholarship benefits or do you have less than 6 months of benefits left?',
+    'ui:widget': 'yesNo',
+    'ui:required': form =>
+      form.isEdithNourseRogersScholarship && form.isEnrolledStem,
+    'ui:options': {
+      expandUnder: 'isEdithNourseRogersScholarship',
+      hideIf: formData => !_.get(formData, 'isEnrolledStem', false),
+    },
+  },
+  isPursuingTeachingCert: {
+    'ui:title':
+      'Do you have a STEM undergraduate degree and are now pursuing a teaching certification?',
+    'ui:widget': 'yesNo',
+    'ui:required': form =>
+      form.isEdithNourseRogersScholarship && !form.isEnrolledStem,
+    'ui:options': {
+      expandUnder: 'isEdithNourseRogersScholarship',
+      hideIf: formData => _.get(formData, 'isEnrolledStem', true),
```

Comment:
This looks very similar to the line I commented on above.

```js
       hideIf: formData => !_.get(formData, 'isEnrolledStem', false), 
      // ...
       hideIf: formData => _.get(formData, 'isEnrolledStem', true), 
```

Can you explain your thought process here?

Reply:
If user answers yes to `isEnrolledStem` then they will have to only answer `view:exhaustionOfBenefits`

If user answers no to `isEnrolledStem` then they will have to answer both `isPursuingTeachingCert` and `view:exhaustionOfBenefitsAfterPursuingTeachingCert`

`view:exhaustionOfBenefits` and `view:exhaustionOfBenefitsAfterPursuingTeachingCert` are the same question however this question should always be asked last 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/5135 has AC and contains links to detailing this flow

Reply:
Got it.  The logic confused me, but I understand the differences now.

Reply:
I spent some additional time thinking about this and came up with some options that might reduce confusion between these two functions, if you wanted to use them.

Lodash seems unnecessary since it is only being used to provide a default, and destructuring could be used for that:

```js
hideIf: ({isEnrolledStem = false}) => !isEnrolledStem,
// ...
hideIf: ({isEnrolledStem = true}) => isEnrolledStem,
```

The destructuring option also isn't _entirely_ clear, so another option would be to make things more explicit:

```js
hideIf: ({ isEnrolledStem }) => !isEnrolledStem || isEnrolledStem === undefined,
// ...
hideIf: ({ isEnrolledStem }) => isEnrolledStem || isEnrolledStem === undefined,
```